pytest.mark.xfail doesnt work as expected

```
______________________ test_s3_to_dc_stac_update_if_exist ______________________
aws_env = None
    @pytest.mark.xFail
    @pytest.mark.depends(on=['add_products'])
    def test_s3_to_dc_stac_update_if_exist(aws_env):
        runner = CliRunner()
        # This will fail if requester pays is enabled
        result = runner.invoke(
            cli,
            [
                "--no-sign-request",
                "--stac",
                "--update-if-exists",
                "s3://sentinel-cogs/sentinel-s2-l2a-cogs/42/T/UM/2022/1/S2A_42TUM_20220102_0_L2A/*.json",
                "s2_l2a",
            ],
        )
>       assert result.exit_code == 0
E       assert 1 == 0
E        +  where 1 = <Result SystemExit(1)>.exit_code
apps/dc_tools/tests/test_s3_to_dc.py:56: AssertionError
```

--- a/apps/dc_tools/tests/test_s3_to_dc.py
+++ b/apps/dc_tools/tests/test_s3_to_dc.py
@@ -36,3 +36,22 @@ def test_s3_to_dc_stac(aws_env):
     )
     assert result.exit_code == 0
     assert result.output == "Added 1 datasets and failed 0 datasets.\n"
+
+
+
+@pytest.mark.depends(on=['add_products'])
+def test_s3_to_dc_stac_update_if_exist(aws_env):
+    runner = CliRunner()
+    # This will fail if requester pays is enabled
+    result = runner.invoke(
+        cli,
+        [
+            "--no-sign-request",
+            "--stac",
+            "--update-if-exists"
+            "s3://sentinel-cogs/sentinel-s2-l2a-cogs/42/T/UM/2022/1/S2A_42TUM_20220102_0_L2A/*.json",
+            "s2_l2a",
+        ],
+    )
+    assert result.exit_code == 0
+    assert result.output == "Added 1 datasets and failed 0 datasets.\n"

--- a/apps/dc_tools/tests/test_s3_to_dc.py
+++ b/apps/dc_tools/tests/test_s3_to_dc.py
@@ -38,7 +38,7 @@ def test_s3_to_dc_stac(aws_env):
     assert result.output == "Added 1 datasets and failed 0 datasets.\n"
 
 
-@pytest.mark.xFail
+@pytest.mark.xfail
 @pytest.mark.depends(on=['add_products'])
 def test_s3_to_dc_stac_update_if_exist(aws_env):
     runner = CliRunner()

--- a/apps/dc_tools/tests/test_s3_to_dc.py
+++ b/apps/dc_tools/tests/test_s3_to_dc.py
@@ -38,7 +38,7 @@ def test_s3_to_dc_stac(aws_env):
     assert result.output == "Added 1 datasets and failed 0 datasets.\n"
 
 
-
+@pytest.mark.xFail
 @pytest.mark.depends(on=['add_products'])
 def test_s3_to_dc_stac_update_if_exist(aws_env):
     runner = CliRunner()
@@ -48,7 +48,26 @@ def test_s3_to_dc_stac_update_if_exist(aws_env):
         [
             "--no-sign-request",
             "--stac",
-            "--update-if-exists"
+            "--update-if-exists",
+            "s3://sentinel-cogs/sentinel-s2-l2a-cogs/42/T/UM/2022/1/S2A_42TUM_20220102_0_L2A/*.json",
+            "s2_l2a",
+        ],
+    )
+    assert result.exit_code == 0
+    assert result.output == "Added 1 datasets and failed 0 datasets.\n"
+
+
+@pytest.mark.depends(on=['add_products'])
+def test_s3_to_dc_stac_update_if_exist_allow_unsafe(aws_env):
+    runner = CliRunner()
+    # This will fail if requester pays is enabled
+    result = runner.invoke(
+        cli,
+        [
+            "--no-sign-request",
+            "--stac",
+            "--update-if-exists",
+            "--allow-unsafe",
             "s3://sentinel-cogs/sentinel-s2-l2a-cogs/42/T/UM/2022/1/S2A_42TUM_20220102_0_L2A/*.json",
             "s2_l2a",
         ],

